graph/src/main/scala/net/spals/dsa/graph/Graph.scala

```diff
@@ -17,10 +17,10 @@ object WeightedGraph {
 
 case class WeightedGraph[T, W](vertices: Set[Vertex[T]], edgeFunc: (Vertex[T], Vertex[T]) => Option[W]) {
 
-  def addEdge(v1: Vertex[T], v2: Vertex[T], w: W): WeightedGraph[T, W] = {
+  def addEdge(v1: Vertex[T], v2: Vertex[T], weight: W): WeightedGraph[T, W] = {
```

Comment:
whats your opinion of variable names in scala? I'm use to java verbose ness by was being simple for functional sake.

not sure if you want vertex1, vertex2, sometimes with pairs I like left/right, I suppose x/y is tradional.